pkg/maps/configmap/configmap.go

```diff
@@ -19,13 +19,176 @@ import (
 	"fmt"
 	"unsafe"
 
+	"github.com/cilium/cilium/common/types"
 	"github.com/cilium/cilium/pkg/bpf"
 	"github.com/cilium/cilium/pkg/logging"
 	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/maps/lxcmap"
 )
 
 var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "map-config")
 
 const (
+	// MapNamePrefix is the basename prefix of endpoint config maps.
 	MapNamePrefix = "cilium_ep_config_"
+
+	// MaxEntries represents the maximum number of elements in the map
+	MaxEntries = 1
+
+	// SkipPolicyIngress causes ingress policy to be skipped.
+	SkipPolicyIngress = 1 << 0
+	// SkipPolicyEgress causes ingress policy to be skipped.
+	SkipPolicyEgress = 1 << 1
+)
+
+var (
+	flagsToString = map[int]string{
+		SkipPolicyIngress: "SKIP_POLICY_INGRESS",
+		SkipPolicyEgress:  "SKIP_POLICY_EGRESS",
+	}
+
+	binMap = map[uint]string{
+		0: "POLICY_INGRESS",
+		1: "POLICY_EGRESS",
+	}
 )
+
+// Flags is a set of endpoint configuration flags interpreted by BPF code.
+//
+// Must be in sync with the enum ep_cfg_flag in <bpf/lib/eps.h>
+type Flags uint32
+
+// String converts the specified flags into a human-readable form.
+func (f Flags) String() string {
+	var buffer bytes.Buffer
+	for i := uint(0); i < 32; i++ {
+		bitVal := int(f & (1 << i))
+		if bitVal > 0 {
+			if flag, ok := flagsToString[bitVal]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+				continue
+			}
+		} else if bitVal == 0 {
+			if flag, ok := binMap[i]; ok {
+				buffer.WriteString(flag)
+				buffer.WriteString(",")
+				continue
+			}
+		}
+		buffer.WriteString(fmt.Sprintf("%04x,", 1<<i))
+	}
+
+	return buffer.String()
+}
+
+// Key is the key used to index into the config map for an endpoint.
+//
+// Must be in sync with the key of CONFIG_MAP in <bpf/lib/maps.h>
+type Key struct {
+	Bits uint32
+}
+
+func (k Key) String() string {
+	return fmt.Sprintf("%d", k)
+}
+
+// GetKeyPtr returns the unsafe pointer to the BPF key
+func (k *Key) GetKeyPtr() unsafe.Pointer { return unsafe.Pointer(k) }
+
+// NewValue returns a new empty instance of the structure representing the BPF
+// map value
+func (k *Key) NewValue() bpf.MapValue { return &EndpointConfig{} }
+
+func (cfg *EndpointConfig) String() string {
+	return fmt.Sprintf("%s,%d,%d,%s,%s,%d,%d,%s", cfg.Flags.String(), cfg.SecurityIdentity, cfg.SecurityIdentityNB, cfg.IPv4.String(), cfg.IPv6.String(), cfg.LXCID, cfg.LXCIDNB, cfg.NodeMAC.String())
```

Comment:
@joestringer I'm ambivalent about whether we want to write all the data here since we aren't plumbing it yet. Open to suggestions.

Reply:
The string output format is not considered stable and is liable to change anyway, so I'd probably just put what we have so far. 

From a CLI consistency point of view, I'd probably want this to be space (tab?) delimited, I think that with tabs and the TabWriter we can end up with nicely formatted columns. Again, just a nit but if you end up changing this anyway then it may be worth considering.

Reply:
I put this as a TODO comment, but added spaces in the meantime.